src/OmniSharp/Api/Navigation/OmnisharpController.FindSymbols.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FindSymbols;
+using OmniSharp.Extensions;
+using OmniSharp.Models;
+
+namespace OmniSharp
+{
+    public partial class OmnisharpController
+    {
+        [HttpPost("findsymbols")]
+        public async Task<QuickFixResponse> FindSymbols(FindSymbolsRequest request = null)
+        {
+            Func<string, bool> isMatch =
+                candidate => request != null
+                ? candidate.IsValidCompletionFor(request.Filter)
+                : true;
+            
+            return await FindSymbols(isMatch);
+        }
+
+        private async Task<QuickFixResponse> FindSymbols(Func<string, bool> predicate)
+        {
+            var symbols = await SymbolFinder.FindSourceDeclarationsAsync(_workspace.CurrentSolution, predicate);
+
+            var quickFixes = (from symbol in symbols
+                              from location in symbol.Locations
+                              where symbol.CanBeReferencedByName
+                                 && symbol.Kind != SymbolKind.Namespace
+                              select ConvertSymbol(symbol, location)).Distinct();
+
+            return new QuickFixResponse(quickFixes);
+        }
+
+        private QuickFix ConvertSymbol(ISymbol symbol, Location location)
+        {
+            var lineSpan = location.GetLineSpan();
+            var path = lineSpan.Path;
+            var documents = _workspace.GetDocuments(path);
+
+            return new QuickFix
+            {
+                Text = new SnippetGenerator().GenerateSnippet(symbol),
```

Comment:
No. See the test output here https://github.com/OmniSharp/omnisharp-roslyn/blob/findsymbols/tests/OmniSharp.Tests/FindSymbolsFacts.cs#L44

I actually thought about renaming this class when I wrote this - the SnippetGenerator has 2 purposes :)